### PR TITLE
replace ncurses-devel OS dependency in CMake easyconfigs using dummy toolchain with proper ncurses dependency

### PR DESCRIPTION
**Pull request by boegel**
These `CMake` easyconfigs using `dummy` toolchain are used as a build dependency for `Eigen` (which uses `cmake` for installation, but it's just a bunch of header files, so nothing gets compiled which means using the `dummy` toolchain is OK).

Requires `ncurses-devel` is stupid/annoying though, we can work around that by including `ncurses` a a *build* dependency and statically linking to it.

It's important that `ncurses` is a *build* dep, otherwise we may run into a version conflict on `ncurses` when these `CMake` easyconfigs are used to build something that has a runtime dependency on (another version of) `ncurses`...

### CODE REVIEW

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in this PR)
node3142.skitty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/2718ceb8f2102b09bd3b91882b19d193 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in this PR)
node2405.golett.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/152e702ecdc569810154ce1491cfe0b3 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in this PR)
node2001.delcatty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/22ab3fc9c32157021a5af8ad8d81904c for a full test report.

**Comment by vanzod** [comment]
Test report by @vanzod
**FAILED**
Build succeeded for 2 out of 6 (5 easyconfigs in this PR)
cermis - Linux debian 9.8, Intel(R) Xeon(R) CPU           X5670  @ 2.93GHz, Python 2.7.13
See https://gist.github.com/9d13a64804510ccfdf5a449b64f30bc8 for a full test report.

**Comment by vanzod** [comment]
Test report by @vanzod
**FAILED**
Build succeeded for 2 out of 6 (6 easyconfigs in this PR)
cermis - Linux debian 9.8, Intel(R) Xeon(R) CPU           X5670  @ 2.93GHz, Python 2.7.13
See https://gist.github.com/48197310858517434d0cd6cf408894b7 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in this PR)
node2119.delcatty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/5fd19f172b586ae9b26c531582c5ed88 for a full test report.

**Comment by vanzod** [comment]
Test report by @vanzod
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in this PR)
cermis - Linux debian 9.8, Intel(R) Xeon(R) CPU           X5670  @ 2.93GHz, Python 2.7.13
See https://gist.github.com/9e43f5d6852d7652838684de4496ca38 for a full test report.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in this PR)
b-an03.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/47269b38d8242f019e4d10f4c2bf873c for a full test report.

File: easybuild/easyconfigs/c/CMake/CMake-3.12.1.eb [hunk 1]
```diff
@@ -11,20 +11,21 @@ description = """
...
+builddependencies = [('ncurses', '6.1')]
 
-configopts = '-- -DCMAKE_USE_OPENSSL=1'
+# Use OS dependencies in order to ensure that CMake can build software that depends on them
+osdependencies = [('openssl-devel', 'libssl-dev', 'libopenssl-devel')]
+
+configopts = "-- -DCMAKE_USE_OPENSSL=1 "
+configopts += "-DCURSES_CURSES_LIBRARY=$EBROOTNCURSES/lib/libcurses.a "
+configopts += "-DCURSES_FORM_LIBRARY=$EBROOTNCURSES/lib/libform.a "
+configopts += "-DCURSES_NCURSES_LIBRARY=$EBROOTNCURSES/lib/libncurses.a -DCURSES_INCLUDE_PATH=$EBROOTNCURSES/include"
```
Comment (akesandgren) [inline easybuild/easyconfigs/c/CMake/CMake-3.12.1.eb]: Hmm, shouldn't there be a " " at the end to be consistent?

File: easybuild/easyconfigs/c/CMake/CMake-3.6.1.eb [hunk 2]
```diff
@@ -7,17 +7,22 @@ homepage = 'http://www.cmake.org'
...
 
 # Use OS dependencies in order to ensure that CMake can build software that
 # depends on them
-osdependencies = [('openssl-devel', 'libssl-dev', 'libopenssl-devel'), ('ncurses-devel', 'libncurses5-dev')]
+osdependencies = [('openssl-devel', 'libssl-dev', 'libopenssl-devel')]
+
+configopts = "-- -DCMAKE_USE_OPENSSL=1 "
+configopts += "-DCURSES_CURSES_LIBRARY=$EBROOTNCURSES/lib/libcurses.a "
+configopts += "-DCURSES_FORM_LIBRARY=$EBROOTNCURSES/lib/libform.a "
+configopts += "-DCURSES_NCURSES_LIBRARY=$EBROOTNCURSES/lib/libncurses.a -DCURSES_INCLUDE_PATH=$EBROOTNCURSES/include"
```
Comment (akesandgren) [inline easybuild/easyconfigs/c/CMake/CMake-3.6.1.eb]: Same here

File: easybuild/easyconfigs/c/CMake/CMake-3.9.1.eb [hunk 3]
```diff
@@ -7,17 +7,22 @@ homepage = 'http://www.cmake.org'
...
 
 # Use OS dependencies in order to ensure that CMake can build software that
 # depends on them
-osdependencies = [('openssl-devel', 'libssl-dev', 'libopenssl-devel'), ('ncurses-devel', 'libncurses5-dev')]
+osdependencies = [('openssl-devel', 'libssl-dev', 'libopenssl-devel')]
+
+configopts = "-- -DCMAKE_USE_OPENSSL=1 "
+configopts += "-DCURSES_CURSES_LIBRARY=$EBROOTNCURSES/lib/libcurses.a "
+configopts += "-DCURSES_FORM_LIBRARY=$EBROOTNCURSES/lib/libform.a "
+configopts += "-DCURSES_NCURSES_LIBRARY=$EBROOTNCURSES/lib/libncurses.a -DCURSES_INCLUDE_PATH=$EBROOTNCURSES/include"
```
Comment (akesandgren) [inline easybuild/easyconfigs/c/CMake/CMake-3.9.1.eb]: And here

File: easybuild/easyconfigs/c/CMake/CMake-3.9.6.eb [hunk 4]
```diff
@@ -7,17 +7,21 @@ homepage = 'http://www.cmake.org'
...
-# Use OS dependencies in order to ensure that CMake can build software that
-# depends on them
-osdependencies = [('openssl-devel', 'libssl-dev', 'libopenssl-devel'), ('ncurses-devel', 'libncurses5-dev')]
+# Use OS dependencies in order to ensure that CMake can build software that depends on them
+osdependencies = [('openssl-devel', 'libssl-dev', 'libopenssl-devel')]
+
+configopts = "-- -DCMAKE_USE_OPENSSL=1 "
+configopts += "-DCURSES_CURSES_LIBRARY=$EBROOTNCURSES/lib/libcurses.a "
+configopts += "-DCURSES_FORM_LIBRARY=$EBROOTNCURSES/lib/libform.a "
+configopts += "-DCURSES_NCURSES_LIBRARY=$EBROOTNCURSES/lib/libncurses.a -DCURSES_INCLUDE_PATH=$EBROOTNCURSES/include"
```
Comment (akesandgren) [inline easybuild/easyconfigs/c/CMake/CMake-3.9.6.eb]: And here

**Comment by akesandgren** [comment]
Going in, thanks @boegel!

**Pull request merged**